src/test/unit_tests/jest_tests/test_config_dir_restructure_upgrade_script.test.js

```diff
@@ -0,0 +1,654 @@
+const DEFAULT_CONF_DIR_PATH = config.NSFS_NC_DEFAULT_CONF_DIR;
+const default_config_fs = new ConfigFS(DEFAULT_CONF_DIR_PATH, config_root_backend, fs_context);
+const hidden_old_accounts_path = path.join(default_config_fs.config_root, `.backup_accounts_dir_${mock_old_version}/`);
+const hidden_access_keys_backup_path = path.join(default_config_fs.config_root, `.backup_access_keys_dir_${mock_old_version}/`);
+
+
+// WARNING:
+// The following test file will check the directory structure created using create_config_dirs_if_missing()
+// which is called when using noobaa-cli, for having an accurate test, it'll be blocked from running on an 
+// env having an existing default config directory and the test executer will be asked to remove the default 
+// config directory before running the test again, do not run on production env or on any env where the existing config directory is being used
+describe('move_old_accounts_dir', () => {
+
+    beforeAll(async () => {
+        await fail_test_if_default_config_dir_exists('test_config_dir_restructure', default_config_fs);
+        await create_config_dir(config.NSFS_NC_DEFAULT_CONF_DIR);
+    });
+
+    afterEach(async () => {
+        await clean_config_dir(default_config_fs);
+    }, TEST_TIMEOUT);
+
+    it('move_old_accounts_dir() - accounts/ dir is missing', async () => {
+        await move_old_accounts_dir(default_config_fs, [], mock_old_version, dbg);
+        await assert_old_account_dir_was_deleted();
+        await assert_backup_dir({});
+    });
+
+    it('move_old_accounts_dir() - accounts/ dir is empty', async () => {
+        await create_fresh_path(default_config_fs.old_accounts_dir_path);
+        await move_old_accounts_dir(default_config_fs, [], mock_old_version, dbg);
+        await assert_old_account_dir_was_deleted();
+        await assert_backup_dir({});
+    });
+
+    it('move_old_accounts_dir() - accounts/ dir contains a single account', async () => {
+        await create_fresh_path(default_config_fs.old_accounts_dir_path);
+        const account_name = 'old_account1';
+        const new_account_data = { _id: '1', name: account_name, user: 'root' };
+        await write_manual_old_account_config_file(default_config_fs, new_account_data);
+        await move_old_accounts_dir(default_config_fs, [account_name], mock_old_version, dbg);
+        await assert_old_account_dir_was_deleted();
+        await assert_backup_dir({ [account_name]: new_account_data});
+    });
+
+    it('move_old_accounts_dir() - accounts/ dir contains 5000 accounts', async () => {
+        await create_fresh_path(default_config_fs.old_accounts_dir_path);
+        const account_ids = Array.from({ length: 5000 }, (_, i) => Number(i + 1));
+        const account_names_obj = {};
+        await P.map_with_concurrency(5000, account_ids, async account_id => {
+            const new_account_data = { _id: String(account_id), name: 'old_account' + account_id, user: 'root' };
```

Comment:
What makes it "old" and "new" (the data is the same, the path is changing)?
```suggestion
            const account_data = { _id: String(account_id), name: 'account' + account_id, user: 'root' };
```